department-of-veterans-affairs/caseflow
# Your Correspondence - In Progress Tab

Resolves [Your Correspondence - In Progress Tab](https://jira.devops.va.gov/browse/APPEALS-35479)

# Description
This PR introduces a functional In Progress tab for the correspondence page. It contains sorting functionality for each of the columns as well as filtering for VA DOR and Task type.

## Acceptance Criteria

## Testing Plan

1. Enter the correspondence queue with user JOLLY_POSTMAN
2. Navigate to the in progress tab.
3. use the up and down arrows on each column to verify that its respective row is being sorted from A-Z or Z-A depending on which toggle you have active (It starts with A-Z)
4. Open the receipt date filter, this has 4 options to test. Between dates, after date, before date, and on date.
5. Fill in the date picker with some data on the table, and ensure that the amount of items within the column are updated to match.
6. Select any of the types of filter available for the task filter, and after checking it, ensure that only that type remained in the table.

# Frontend
## User Facing Changes

 BEFORE|AFTER
 
<img width="1285" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110805785/99c9c438-c8b4-43f2-9831-12af79a4190f">
|
<img width="1301" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110805785/f6c2fe3c-2f32-49b2-b0cd-f1f9d0afb418">

code comments at the top of the class, module, and/or component.

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?
